File: HMSProjectOfMine/HMSProjectOfMine/DTOs/PatientDTO.cs
using HMSProjectOfMine.Enums;
using HMSProjectOfMine.Models;

namespace HMSProjectOfMine.DTOs
{
    public class PatientCreateDto
    {
        public string PatientNo { get; set; } = null!;
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public int? RegistrationId { get; set; }
        public Gender Gender { get; set; }
        public int Age { get; set; }
        public PatientType PatientType { get; set; }
        public VisitType? VisitType { get; set; }

        // Token information to be generated upon patient creation
        public decimal TokenFee { get; set; }
        public int DoctorId { get; set; }
        public int ChamberId { get; set; }
    }

    public class PatientReadDto
    {
        public int PatientId { get; set; }
        public string PatientNo { get; set; } = null!;
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public int? RegistrationId { get; set; }
        public Gender Gender { get; set; }
        public int Age { get; set; }
        public DateTime FirstVisitDate { get; set; }
        public PatientType PatientType { get; set; }
        public VisitType? VisitType { get; set; }
        public virtual Token Tokens { get; set; }

    }

    public class PatientUpdateDto
    {
        public int PatientId { get; set; }
        public string? PatientNo { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public int? RegistrationId { get; set; }
        public Gender? Gender { get; set; }
        public int? Age { get; set; }
        public PatientType? PatientType { get; set; }
        public VisitType? VisitType { get; set; }
    }

    public class TokenReadDto
    {
        public int TokenId { get; set; }
        public string TokenNumber { get; set; } = null!;
        public decimal TokenFee { get; set; }
        public DateTime IssueTime { get; set; }
        public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public int ChamberId { get; set; }
    }

    public class PatientWithTokenDto : PatientReadDto
    {
        public TokenReadDto? Token { get; set; }
    }

    
}

File: HospitalManagement/src/app/models/PatientDTO.ts
export interface PatientCreate {
  patientNo: string;
  firstName: string;
  lastName: string;
  registrationId?: number;
  gender: Gender; // Or an enum if you prefer in Angular
  age: number;
  patientType: PatientType; // Or an enum
  visitType?: VisitType;
  tokenFee: number;
  doctorId: number;
  chamberId: number;
}

export enum Gender {
  Male = 'Male',
  Female = 'Female',
  Other = 'Other',
}

export enum VisitType {
  FirstVisit = 'FirstVisit',
  FollowUp = 'FollowUp',
  TestReview = 'TestReview',
  Others = 'Others',
}

export enum PatientType {
  Indoor = 'Indoor',
  Outdoor = 'Outdoor',
}

export interface PatientRead {
  patientId: number;
  patientNo: string;
  firstName: string;
  lastName: string;
  registrationId?: number;
  gender: string;
  age: number;
  firstVisitDate: Date;
  patientType: string;
  visitType?: string;
}

export interface PatientUpdate {
  patientId: number;
  patientNo?: string;
  firstName?: string;
  lastName?: string;
  registrationId?: number;
  gender?: Gender;
  age?: number;
  patientType?: PatientType;
  visitType?: VisitType;
}

export interface TokenRead {
  tokenId: number;
  tokenNumber: string;
  tokenFee: number;
  issueTime: Date;
  patientId: number;
  doctorId: number;
  chamberId: number;
}

export interface PatientWithToken extends PatientRead {
  token?: TokenRead;
}

export interface DoctorDropdown {
  doctorId: number;
  firstName: string;
  lastName: string;
  specializationName: string;
  departmentName: string;
}

export interface ChamberDropdown {
  chamberId: number;
  chamberNo: string;
  location: string;
}
